op-chain-ops/script/check_super_root.go

```
@@ -0,0 +1,379 @@
+	m.log.Info("Starting super root calculation process")
+
+	// Use the provided context for all operations
+	if err := m.initClientsAndFetchIDs(ctx); err != nil {
+		return fmt.Errorf("failed to initialize clients: %w", err)
+	}
+
+	if err := m.findAnchorTimestamp(ctx); err != nil {
+		return fmt.Errorf("failed to find anchor timestamp: %w", err)
+	}
+	m.log.Info("Found common anchor timestamp", "timestamp", m.anchorTimestamp)
+
+	if err := m.deriveParamsAndFindTargetBlocks(ctx); err != nil {
+		return fmt.Errorf("failed to derive parameters and find target blocks: %w", err)
+	}
+
+	if err := m.calculateOutputRoots(ctx); err != nil {
+		return fmt.Errorf("failed to calculate output roots: %w", err)
+	}
+
+	if err := m.calculateSuperRoot(); err != nil {
+		return fmt.Errorf("failed to calculate super root: %w", err)
+	}
+
+	m.printResults()
+
+	m.log.Info("Super root calculation process finished successfully")
+	return nil
+}
+
+// initClientsAndFetchIDs establishes connections to all RPC endpoints
+// and retrieves their chain IDs.
+func (m *SuperRootMigrator) initClientsAndFetchIDs(ctx context.Context) error {
+	m.log.Info("Initializing clients and fetching chain IDs...")
+	endpointsToRemove := []string{}
+
+	for _, endpoint := range m.rpcEndpoints {
+		m.log.Debug("Dialing RPC endpoint", "url", endpoint)
+		client, err := ethclient.DialContext(ctx, endpoint)
+		if err != nil {
+			m.log.Error("Failed to connect to RPC endpoint, skipping", "url", endpoint, "err", err)
+			endpointsToRemove = append(endpointsToRemove, endpoint)
+			continue
+		}
+		m.ethClients[endpoint] = client
+		m.log.Info("Connected to client", "url", endpoint)
+
+		chainID, err := client.ChainID(ctx)
+		if err != nil {
+			m.log.Error("Failed to get chain ID, closing client and skipping", "url", endpoint, "err", err)
+			client.Close()
+			delete(m.ethClients, endpoint)
+			endpointsToRemove = append(endpointsToRemove, endpoint)
+			continue
+		}
```

If a supplied endpoint fails, the program should report that an error and exit. We don't want to wind up ignoring an endpoint and then creating an invalid super root.

---

Yeah good call! Updated here [4583440](https://github.com/ethereum-optimism/optimism/pull/15514/commits/45834400347483fea61308765d589607d7b0454c)